### PR TITLE
fix(lang-rust): Custom dockerfile issue for cargo-install to non-existent /workspace dir

### PR DESCRIPTION
**Pull request by axonasif**
## Description

CARGO specific env var modifications are now done inside workspace-session but not for docker env.

## Related Issue(s)

Fixes [this on discord](https://canary.discord.com/channels/816244985187008514/946731644465852466/946805048116121640)

## Release Notes

```release-note
```

### CODE REVIEW

File: chunks/lang-rust/Dockerfile [hunk 1]
```diff
@@ -18,17 +18,12 @@ RUN cp $HOME/.profile $HOME/.profile_orig \
...
     && printf '%s\n' '#!/usr/bin/bash -eu' \
                         'exec env -u CARGO_HOME "$(command -v rustup.main)" "$@"' > "$_rustup_path" \
-    && chmod 0755 "$_rustup_path"
+    && chmod 0755 "$_rustup_path" && rm $HOME/.profile_orig
 
 RUN cargo install cargo-watch cargo-edit cargo-workspaces
-ENV CARGO_HOME=/workspace/.cargo
-ENV PATH=$CARGO_HOME/bin:$PATH
-
-RUN sudo mkdir -p $CARGO_HOME \
```
Comment (sagor999) [inline chunks/lang-rust/Dockerfile:31]: Just double checking, is it intentional to remove `chown` here?

File: chunks/lang-rust/Dockerfile [hunk 2]
```diff
@@ -18,17 +18,12 @@ RUN cp $HOME/.profile $HOME/.profile_orig \
...
-    && sudo chown -R gitpod:gitpod $CARGO_HOME
-
-RUN bash -lc "cargo install cargo-watch cargo-edit cargo-tree cargo-workspaces"
```
Comment (sagor999) [inline chunks/lang-rust/Dockerfile:34]: And also intentional to remove this `cargo install` here?

Comment (axonasif) [reply chunks/lang-rust/Dockerfile:34]: Yes, please see my previous comment.

File: chunks/lang-rust/Dockerfile (hunk 1 above)
Comment (axonasif) [reply chunks/lang-rust/Dockerfile:31]: Yes, I'm not sure how these lines got added by themselves again as I had removed them in a PR previously https://github.com/gitpod-io/workspace-images/pull/587/files

**Review by sagor999 — APPROVED** [review]

**Pull request merged**